src/Interface/Modules/Render/ViewScene.h

```diff
@@ -74,7 +74,7 @@ namespace SCIRun {
 
     protected Q_SLOTS:
       //---------------- New Geometry --------------------------------------------------------------
-      void newGeometryValue();
+      void newGeometryValue(bool force = false);
```

Comment:
This is a first review that will cover style issues. I haven't run the code yet. 

For `newGeometryValue` changes, we already know it's an overly complicated function. Adding a parameter with default value just makes it more complicated. The trick to smoothing this out is as follows:
- [ ] Remove the default value `= false`
- [ ] Rename the parameter from `force` to be as explicit as possible--in this case something like `forceImmediateGarbageCollection` 
- [ ] Add an overloaded `newGeometryValue` with no parameters that just calls the parameter one, passing false. Then you don't have to change exisiting usages like before, and the implementation stays in the .cc file

Reply:
Then, perhaps, you can have a newly-named function that calls `newGeometryValue` passing true, like `newGeometryValueWithCleanup` or something. See if there is a name that makes sense there. Reading the client code will be a lot easier than having to remember what the magic boolean does.